vizro-core/examples/visual-vocabulary/custom_charts.py

```diff
@@ -310,3 +310,59 @@ def diverging_stacked_bar(data_frame: pd.DataFrame, **kwargs) -> go.Figure:
         fig.add_hline(y=0, line_width=2, line_color="grey")
 
     return fig
+
+
+@capture("graph")
+def lollipop(data_frame: pd.DataFrame, **kwargs):
+    """Creates a lollipop based on px.scatter.
+
+    A lollipop chart is a variation of a bar chart where each data point is represented by a line and a dot at the end
+    to mark the value.
+
+    Inspired by: https://community.plotly.com/t/how-to-make-dumbbell-plots-in-plotly-python/47762
+
+    Args:
+        data_frame: DataFrame for the chart. Can be long form or wide form.
+            See https://plotly.com/python/wide-form/.
+        **kwargs: Keyword arguments to pass into px.scatter (e.g. x, y, labels).
+            See https://plotly.com/python-api-reference/generated/plotly.scatter.html.
+
+    Returns:
+        go.Figure: Lollipop chart.
+    """
+    # Unlike the column_and_line chart, where all traces hold equal significance, here the traces differ in importance.
+    # The primary scatter plot is the main trace, while the additional traces merely serve as connecting lines.
+    # Therefore, should we apply the kwargs solely to the main scatter plot, as illustrated below?
+    fig = px.scatter(data_frame, **kwargs)
+
+    # Enable for both orientations
+    orientation = fig.data[0].orientation
+    x_array = fig.data[0]["x"]
+    y_array = fig.data[0]["y"]
+
+    for i in range(len(data_frame)):
+        fig.add_trace(
+            go.Scatter(
+                x=[0, x_array[i]] if orientation == "h" else [x_array[i], x_array[i]],
+                y=[y_array[i], y_array[i]] if orientation == "h" else [0, y_array[i]],
+                mode="lines",
+            )
+        )
+
+    if orientation == "h":
+        yaxis_showgrid = False
+        xaxis_showgrid = True
+    else:
+        yaxis_showgrid = True
+        xaxis_showgrid = False
```

Comment:
I can't remember do we aim for the optimised code or for the readability in vivivo examples? However, there is a half-optimised suggestion.

For the further optimisation we can calculate `x_coords` and `y_coords` with the `x_or_y` technique that's used within many other vivivo examples. (I like "vivivo" abbreviation 😄)

(If you accept the suggestion, apply the same changes in the `examples/lollipop.py` too)

```suggestion
    # Enable for both orientations
    is_horizontal = fig.data[0].orientation == "h"

    x_coords = [[0, x] if is_horizontal else [x, x] for x in fig.data[0]["x"]]
    y_coords = [[y, y] if is_horizontal else [0, y] for y in fig.data[0]["y"]]

    for x, y in zip(x_coords, y_coords):
        fig.add_trace(go.Scatter(x=x, y=y, mode="lines"))

    xaxis_showgrid = is_horizontal
    yaxis_showgrid = not is_horizontal
```

Reply:
Ah this is great, thanks! ⭐